backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java

```diff
@@ -1907,4 +1911,81 @@ public ProjectData searchByTag(String tag, User user) {
     public ProjectData searchByType(String type, User user) {
         return repository.searchByType(type, user);
     }
+
+    public ByteBuffer getReportDataStream(List<Project> documents, User user, String exporterObject, boolean extendedByReleases)
+            throws SW360Exception, TException {
+        ThriftClients thriftClients = new ThriftClients();
+        InputStream stream = null;
+        switch (exporterObject) {
+        case PROJECTS:
+            ProjectExporter exporter = getProjectExporterObject(documents, user, extendedByReleases, thriftClients);
+            try {
+                stream = exporter.makeExcelExport(documents);
+            } catch (SW360Exception e) {
+                e.printStackTrace();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            break;
+
+        default:
+            break;
+        }
+        try {
+            return ByteBuffer.wrap(IOUtils.toByteArray(stream));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    private ProjectExporter getProjectExporterObject(List<Project> documents, User user, boolean extendedByReleases,
+            ThriftClients thriftClients) throws SW360Exception {
+        ProjectExporter exporter = new ProjectExporter(thriftClients.makeComponentClient(),
+                thriftClients.makeProjectClient(), user, documents, extendedByReleases);
+        return exporter;
+    }
+
+    public String getReportInEmail(List<Project> documents, User user, String exporterObject,
+            boolean extendedByReleases) throws SW360Exception, TException {
+        ThriftClients thriftClients = new ThriftClients();
+        String result = null;
+        switch (exporterObject) {
+        case PROJECTS:
+            ProjectExporter exporter = getProjectExporterObject(documents, user, extendedByReleases, thriftClients);
+            try {
+                result = exporter.makeExcelExportForProject(documents, user);
+            } catch (SW360Exception e) {
```

Comment:
Either catch it with appropriate log message instead of `printStackTrace()` or throw it

Reply:
Resolved